src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+public class Duke {
+
+    private Storage storage;
+    private Ui ui;
+    private TaskList tasks;
+    
+    static String divider = "____________________________________________________________";
+    
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.load());
+    }
+
+
+    /**
+     * Adds a task to storage.
+     * @param task 
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+        storage.save(tasks);
+    }
+
+    /**
+     * Deletes task of the index from storage.
+     * @param index
+     */
+    public void deleteTask(int index) {
+        tasks.delete(index);
+        storage.save(tasks);
+    }
+
+    /**
+     * Marks a task as done and saves it.
+     * @param index
+     */
+    public void mark(int index) {
+        tasks.mark(index);
+        storage.save(tasks);
+    }
+
+    /**
+     * Unmark a task as undone and saves it.
+     * @param index
+     */
+    public void unmark(int index) {
+        tasks.unmark(index);
+        storage.save(tasks);
+    }
+    
+    public void run() {
+        ui.sendStartMessage();
+        Scanner input = new Scanner(System.in);
+        while (true) {
+            String text = input.nextLine();
+            if (text.equals("bye")) {
+                ui.sendByeMessage();
```

Comment:
Perhaps you could encapsulate each of these commands into a Command object so that less of the logic is present in the main Duke.java code